var/spack/repos/builtin/packages/py-rios/package.py

```diff
@@ -27,10 +31,25 @@ class PyRios(PythonPackage):
     version("1.4.11", sha256="b7ae5311f987b32f1afe1fabc16f25586de8d15c17a69405d1950aeada7b748e")
     version("1.4.10", sha256="6324acccc6018f9e06c40370bc366dc459890e8c09d26e0ebd245f6fd46dad71")
 
-    variant("parallel", default=True, description="Enables the parallel processing module")
+    # In 1.4.x, parallel processing was an extra add-on
+    variant(
+        "parallel",
+        default=True,
+        when="@1.4.16:1.4.99",
+        description="Enables the 1.4.x parallel processing module (deprecated)",
+    )
+    # In 2.x, there is substantial concurrency always built-in, but using it
+    # across multiple machines requires an extra dependency.
+    variant(
+        "multimachine",
+        default=False,
+        when="@2.0.0:",
+        description="Enable compute worker kinds that run across multiple machines",
+    )
 
     # pip silently replaces distutils with setuptools
     depends_on("py-setuptools", type="build")
     depends_on("py-numpy", type=("build", "run"))
     depends_on("gdal+python", type=("build", "run"))
-    depends_on("py-cloudpickle", type=("build", "run"), when="@1.4.16:+parallel")
+    depends_on("py-cloudpickle", type="run", when="@1.4.16:1.4.99+parallel")
+    depends_on("py-cloudpickle", type="run", when="@2.0.0:+multimachine")
```

Comment:
Same here about simplifying the version ranges.

Reply:
Thanks for those. I had not realised the simpler forms would be sufficient :-)